build/config/compiler/compiler.gni

```diff
@@ -43,7 +43,7 @@ declare_args() {
   c_standard = "gnu11"
 
   # C++ standard level (value for -std flag).
-  if (current_os == "android") {
+  if (current_os == "linux" || current_os == "mac" || current_os == "android") {
```

Comment:
That said, this will make it _really_ easy to write code locally that depends on c++17 that then fails in CI (e.g. using any of the template metaprogramming improvements)....  Is the plan to move to c++17 for all platforms?  If not, we should strongly consider setting c++14 explicitly for "core" bits for now (src/app most importantly.

Reply:
plan is everything C++17. Asked platform DRIs and everyone said yes except the ones who did not reply. Several embedded platforms are already C++17 in places where pigweed rpc is used because apparently that is a requirement.

We may suffer with unsupported bits for a short while, however I do expect C++17 move to happen completely.

Reply:
I suspect the CI surprise is probably par for the course for now. c++14 is now 10 years old, we need to pull the bandaid off at some point.

Reply:
To be clear, I am _very_ much in favor of moving to c++17 so I can stop all the ::value bits and use _t on the ends of things instead.  And if constexpr.  And so on.  ;)

But because I want to use those things, I expect lots of breakage in my personal future until we move everything over is all.

Reply:
Fair enough, do you think it would be better to just swap all the platforms and deal with the non-c++17 compliant platforms? Given that all but one have responded, maybe that's reasonable.

Reply:
I think it's fine to do this either way; I just want to make sure we don't get stuck in a long limbo of some platforms being one way and some another.